src/backend/distributed/commands/utility_hook.c

```diff
@@ -852,6 +860,33 @@ PlanAlterOwnerStmt(AlterOwnerStmt *stmt, const char *queryString)
 }
 
 
+/*
+ * PlanAlterObjectDependsStmt gets called during the planning phase for
+ * ALTER ... DEPENDS ON EXTENSION ... statements. Based on the object type we call out to
+ * a specialized implementation. If no implementation is available we do nothing, eg. we
+ * allow the local node to execute.
+ */
+static List *
+PlanAlterObjectDependsStmt(AlterObjectDependsStmt *stmt, const char *queryString)
```

Comment:
I find it little confusing that a function that is either going to error out or do nothing returns a `List`.

In general, we make this functions in a way that `ErrorIfAlterDistributedFunctionDependsOnExtension()` to make things a lot more explicit.

I think your point would be that this is consistent with other similar functions, which is fair. Still, I've a slight preference for changing the function name/return value such that it becomes easier to follow the code.

What's your take on this?

Reply:
I agree that only looking at this function might make it a bit strange that the function signature is `List * Plan...(... stmt, const char *queryString)` if its only function is to return an error if it is a distributed function.

But if you look at it in the broader scope of how the utility hook is structured it is a prelude to a structured way of responding to statements from citus' perspective.

```C
ddlJobs = PlanStmtType(stmt, queryString);
ExecuteOriginalUtilityHook(...)
ProcessStmtType(stmt, queryString);
ExecuteDDLJobsOnWorkers(ddlJobs);
```

Currently the logic is not as clear as above due to the history of how the utility hook came to be. However there is a clear interface forming now where based on the type of statement we need to retrieve two (potentially `NULL`) function pointers, eg. `PlanStmt` and `ProcessStmt`. Hence the current setup where it both returns a `List *` and takes a `queryString`.

Even if we don't have the abstraction described here it becomes easier to read and understand the body of our utility hook if the functions we call look a lot alike. Once you understand the flow of one you understand them all and it becomes natural to navigate. If we would tailer the names of the functions and the input arguments specific to each situation my fear would be that the utility hook is not understandable to the same degree.